File: Data Structures & Algorithms/string-encode-and-decode/submission-2.java
class Solution {

    public String encode(List<String> strs) {
        String secret = "";
    
        for(String s : strs) {
            secret += s.length() + "#" + s;
        }
        return secret;
    }

    public List<String> decode(String str) {
       List<String> wordList = new ArrayList<>();
    int startWord = 0;

    while(startWord < str.length()) {
      int pointer = startWord;

      while(str.charAt(pointer) != '#') {
        pointer++;
      }

      System.out.println("startWord: " + startWord + " Pointer: " + pointer);
      int wordLength = Integer.parseInt(str.substring(startWord, pointer));

      String temp = "";
      temp = str.substring(pointer + 1, pointer + 1 + wordLength);
      wordList.add(temp);
      
      startWord = pointer + 1 + wordLength;

    }
    
    return wordList;
    
    }
}

File: Data Structures & Algorithms/length-of-last-word/submission-2.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        s = s.strip()
        pointer = len(s) - 1

        if(len(s) == 1):
            return 1

        while(s[pointer] != " "):
            pointer = pointer - 1

        return (len(s) - 1 - pointer)

            
